src/components/Cart.js

```diff
@@ -1,108 +1,114 @@
+        }
+    };
 
-      <div className="grid grid-cols-3 gap-4">
-        {cart.map((item) => (
-          <div
-            key={item?.card?.info?.id}
-            className="bg-white shadow-md p-4 rounded-md"
-          >
-            <img
-              src={
-                "https://media-assets.swiggy.com/swiggy/image/upload/fl_lossy,f_auto,q_auto,h_600/" +
-                item?.card?.info?.imageId
-              }
-              alt={item?.card?.info?.name}
-              className="w-32 h-32 object-cover rounded-md mb-2"
-            />
-            <h3 className="text-xl font-bold mb-2">{item?.card?.info?.name}</h3>
-            <p className="text-lg mb-2">{item?.card?.info?.description}</p>
-            <p className="text-lg mb-2">
-              Price: ₹
-              {(item?.card?.info?.price || item?.card?.info?.defaultPrice) /
-                100}
-            </p>
+    const totalPrice = () => {
+        return cart.reduce((total, item) => {
+            const itemPrice = item?.card?.info?.price || item?.card?.info?.defaultPrice;
+            return total + (itemPrice * quantities[item?.card?.info?.id]) / 100; // Ensure price is in the correct format
+        }, 0);
+    };
 
-            <div className="flex items-center mb-4">
-              <button
-                className="bg-blue-500 text-white px-2 py-1 rounded-md"
-                onClick={() => dispatch(removeFromCart(item))}
-              >
-                -
-              </button>
-              <span className="mx-4 text-lg">
-                {
-                  cart?.find((e) => e?.card?.info?.id === item?.card?.info?.id)
-                    ?.quantity
-                }
-              </span>
-              <button
-                className="bg-blue-500 text-white px-2 py-1 rounded-md"
-                onClick={() => dispatch(addToCart(item))}
-              >
-                +
-              </button>
+    if (cart.length === 0) {
+        return (
+            <div className="flex justify-center items-center h-screen">
+                <h1 className="text-3xl font-bold">Your Cart is Empty</h1>
             </div>
+        );
+    }
 
+    return (
+        <div className="mx-2 px-2">
+            <h1 className="text-3xl font-bold mb-4">Cart</h1>
             <button
-              className="bg-red-500 text-white px-4 py-2 rounded-md"
-              onClick={() => dispatch(removeItemFromCart(item))}
+                className={`bg-red-500 text-white px-4 py-2 rounded-md mb-4 ${cart.length === 0 && "opacity-50 cursor-not-allowed"}`}
+                onClick={handleClearCart}
+                disabled={cart.length === 0}
             >
-              Remove
+                Clear Cart
             </button>
-          </div>
-        ))}
-      </div>
 
-      <div className="mt-4">
-        <h2 className="text-2xl font-bold">Total: ₹{totalPrice() / 100}</h2>
-      </div>
-    </div>
-  );
+            <div className="grid grid-cols-3 gap-4">
```

Comment:
it does not reflecting when i increase item amount from menu itself , it working individually which is wrong

[live site](https://loco-musica.vercel.app/cart)

go to site try add item to chart then click to + icon to increase
it is reflecting in cart ,

but with your it is not reflecting in card

Reply:
okay I will reupdate it with your code only